integration-tests/jiva/node-stickiness/replicadeletion/jiva_replica_deletion_suit_test.go

```diff
@@ -0,0 +1,153 @@
+	// need to run this test
+	minNodeCount int = 3
+	// parentDir is the OpenEBS artifacts source directory
+	parentDir artifacts.ArtifactSource = "../../"
+	// jiva-test namespace to deploy jiva ctrl & replicas
+	nameSpaceYaml artifacts.Artifact = `
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: jiva-test
+`
+)
+
+var (
+	//Client set
+	defaultComponentsInstaller []*installer.DefaultInstaller
+)
+
+func TestSource(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Pod")
+}
+
+func init() {
+	flag.Parse()
+}
+
+// TODO: Refactor below code based on the framework changes
+// getPodList returns the list of running pod object
+func getPodList(namespace, lselector string, podCount int) (pods *corev1.PodList) {
+	// Verify phase of the pod
+	var err error
+	Eventually(func() int {
+		pods, err = pod.
+			KubeClient(pod.WithNamespace(namespace)).
+			List(metav1.ListOptions{LabelSelector: lselector})
+		Expect(err).ShouldNot(HaveOccurred())
+		return pod.
+			ListBuilder().
+			WithAPIList(pods).
+			WithFilter(pod.IsRunning()).
+			List().
+			Len()
+	},
+		defaultTimeOut, defaultPollingInterval).
+		Should(Equal(podCount), "Pod count should be "+string(podCount))
+	return
+}
+
+var _ = BeforeSuite(func() {
+	// Fetching the kube config path
+	configPath, err := kubernetes.GetConfigPath()
+	Expect(err).ShouldNot(HaveOccurred())
+
+	// Setting the path in environemnt variable
+	err = os.Setenv(string(v1alpha1.KubeConfigEnvironmentKey), configPath)
+	Expect(err).ShouldNot(HaveOccurred())
+
+	// Check the running node count
+	nodes, err := node.
+		KubeClient().List(metav1.ListOptions{})
+	Expect(err).ShouldNot(HaveOccurred())
+	nodeCnt := node.
+		NewListBuilder().
+		WithAPIList(nodes).
+		WithFilter(node.IsReady()).
+		List().
+		Len()
+	Expect(nodeCnt).Should(Equal(minNodeCount), "Running node count should be "+string(nodeCnt))
+
+	// Fetch openebs component artifacts
+	openebsartifacts, errs := artifacts.GetArtifactsListUnstructuredFromFile(parentDir + artifacts.OpenEBSArtifacts)
+	Expect(errs).Should(HaveLen(0))
+
+	By("Installing OpenEBS components")
+	// Installing the artifacts to kubernetes cluster
+	for _, artifact := range openebsartifacts {
+		defaultInstaller, err := installer.BuilderForObject(artifact).Build()
+		Expect(err).ShouldNot(HaveOccurred())
+		err = defaultInstaller.Install()
+		Expect(err).ShouldNot(HaveOccurred())
+		defaultComponentsInstaller = append(defaultComponentsInstaller, defaultInstaller)
+	}
+	// Creates jiva-test namespace
+	testNamespaceArtifact, err := artifacts.GetArtifactUnstructured(artifacts.Artifact(nameSpaceYaml))
+	Expect(err).ShouldNot(HaveOccurred())
+	namespaceInstaller, err := installer.BuilderForObject(testNamespaceArtifact).Build()
+	Expect(err).ShouldNot(HaveOccurred())
+	err = namespaceInstaller.Install()
+	Expect(err).ShouldNot(HaveOccurred())
+	defaultComponentsInstaller = append(defaultComponentsInstaller, namespaceInstaller)
+
+	By("Started deploying OpenEBS components")
```

Comment:
These all logic should be done via the framework. An XYZ test suite should not bother about install, uninstall & verification of openebs components. It should instruct the framework to handle.

```go
// In Before Suite
f, err := framework.Builder().
  WithOpenEBSVersion("0.8.2").
  WithPlatform("openshift").
  WithPlatformVersion("1.11.1").
  WithOperatorInstall(false).  // indicates environment is already setup
  WithCheck(framework.IsDeployedEventually()).
  Build()
if err != nil {
  // ...
}

f.Install()

// In After Suite
f.Uninstall()
```

Reply:
It will be with `frame work` PR